server/data/redisClient.ts

```diff
@@ -5,16 +5,57 @@ import config from '../config'
 
 export type RedisClient = ReturnType<typeof createClient>
 
-export const createRedisClient = (): RedisClient => {
-  const client = createClient({
-    port: config.redis.port,
+class RedisClientWithPrefix {
+  private client: ReturnType<typeof createClient>
+
+  private prefix: string
+
+  constructor(client: ReturnType<typeof createClient>, prefix = 'systemToken:') {
+    this.client = client
+    this.prefix = prefix
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public async set(key: string, value: string, options?: any): Promise<void> {
+    await this.client.set(this.prefix + key, value, options)
+  }
+  /* eslint-enable @typescript-eslint/no-explicit-any */
+
+  public async get(key: string): Promise<string | null> {
+    return this.client.get(this.prefix + key)
+  }
+
+  public async del(key: string): Promise<void> {
+    await this.client.del(this.prefix + key)
+  }
+
+  public getOriginalClient() {
+    return this.client
+  }
+}
+
+const url =
+  config.redis.tls_enabled === 'true'
+    ? `rediss://${config.redis.host}:${config.redis.port}`
+    : `redis://${config.redis.host}:${config.redis.port}`
+
+export const createRedisClient = (): RedisClientWithPrefix => {
+  const client: RedisClient = createClient({
+    url,
     password: config.redis.password,
-    host: config.redis.host,
-    tls: config.redis.tls_enabled === 'true' ? {} : false,
-    prefix: 'systemToken:',
+    socket: {
+      reconnectStrategy: (attempts: number) => {
+        // Exponential back off: 20ms, 40ms, 80ms..., capped to retry every 30 seconds
+        const nextDelay = Math.min(2 ** attempts * 20, 30000)
+        logger.info(`Retry Redis connection attempt: ${attempts}, next attempt in: ${nextDelay}ms`)
+        return nextDelay
+      },
+    },
   })
 
   client.on('error', (e: Error) => logger.error('Redis client error', e))
 
-  return client
+  client.connect().catch((e: Error) => logger.error('Failed to connect to Redis', e))
```

Comment:
maybe do this in the setUpWebSession [like in the template project](https://github.com/ministryofjustice/hmpps-template-typescript/blob/576409caf776c876219b897c715dc329ca30add4/server/middleware/setUpWebSession.ts#L13)